src/cliffordsynthesis/HeuristicStrategy.cpp

```diff
@@ -0,0 +1,127 @@
+                }
+                for (auto* t: threads) {
+                    t->join();
+                }
+                for (auto* t: threads) {
+                    delete t;
+                }
+                for (auto* r: results) {
+                    if (r->result == logicbase::Result::UNSAT) {
+                        totalResult.result = logicbase::Result::UNSAT;
+                        break;
+                    }
+                }
+                if (totalResult.result == logicbase::Result::UNSAT) {
+                    DEBUG() << "UNSAT, increasing split size." << std::endl;
+                    split += std::max(1.0, split * 0.2);
+                    circuitSplit += std::max(1.0, circuitSplit * 0.2);
+                    if (circuitSplit > configuration.targetCircuit.getNindividualOps()) {
+                        stopping = true;
+                        break;
+                    }
+                    break;
+                }
+            }
+            for (auto* r: results) {
+                if (r->resultStringCircuit.empty()) {
+                    continue;
+                }
+                qc::QuantumComputation splitResult;
+                std::istringstream     iss(r->resultStringCircuit);
+                splitResult.import(iss, qc::OpenQASM);
+                for (const auto& gate: splitResult) {
+                    localResultCircuit.insert(localResultCircuit.end(),
+                                              gate->clone());
+                }
+                delete r;
+            }
+            if (totalResult.result == logicbase::Result::SAT) {
+                Tableau resultingTableau{localResultCircuit};
+                DEBUG() << "Equality (Results): "
+                        << ((fullTableau == resultingTableau) ? "True" : "False")
+                        << std::endl;
+                DEBUG() << "Original Circuit size: " << circuit.getNindividualOps()
+                        << std::endl;
+                DEBUG() << "Optimized Circuit size: "
+                        << localResultCircuit.getNindividualOps() << std::endl;
+                TRACE() << "Resulting Circuit: " << std::endl;
+                std::ostringstream ss;
+                localResultCircuit.dump(ss, qc::Format::OpenQASM);
+                TRACE() << ss.str() << std::endl;
+                auto                          end  = std::chrono::high_resolution_clock::now();
+                std::chrono::duration<double> diff = end - start;
+                INFO() << "Time for complete run: " << diff.count() << std::endl;
+                if (circuit.getNindividualOps() ==
+                    localResultCircuit.getNindividualOps()) {
+                    break;
+                }
+                circuit = localResultCircuit.clone();
+            }
+        }
+        std::stringstream ss;
+        circuit.dump(ss, qc::Format::OpenQASM);
+        synthesizer.optimalResults.resultStringCircuit = ss.str();
```

Comment:
there is a function for that somewhere in the newly added coed

Reply:
Fixed